src/Stack/Ghci.hs

```diff
@@ -215,12 +218,22 @@ ghciSetup mainIs stringTargets = do
                                  return (Just (name, (cabalfp, simpleTargets)))
                              Nothing -> return Nothing
                     else return Nothing
+    -- Try to build, but optimistically launch GHCi anyway if it fails (#1065)
+    case (mbuildFirst, M.null realTargets) of
```

Comment:
I think the check for (null realTargets) should be removed.  See point 1 on #1249 for why.  See also #817 for why the check was there in the first place.